app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/localService.ts

```diff
@@ -0,0 +1,157 @@
+import * as cloudService from './cloudService'
+import * as newtype from '../newtype'
+import * as platformModule from '../platform'
+import * as projectManager from './projectManager'
+
+// ========================
+// === Helper functions ===
+// ========================
+
+function ipWithSocketToAddress(ipWithSocket: projectManager.IpWithSocket) {
+    return newtype.asNewtype<cloudService.Address>(`ws://${ipWithSocket.host}:${ipWithSocket.port}`)
+}
+
+// ===============
+// === Backend ===
+// ===============
+
+interface CurrentlyOpenProjectInfo {
+    id: projectManager.ProjectId
+    project: projectManager.OpenProject
+}
+
+export class Backend implements Partial<Omit<cloudService.Backend, 'platform'>> {
+    readonly platform = platformModule.Platform.desktop
+    private readonly projectManager = projectManager.ProjectManager.default()
+    private currentlyOpeningProjectId: string | null = null
+    private currentlyOpenProject: CurrentlyOpenProjectInfo | null = null
+
+    async listDirectory(): Promise<cloudService.Asset[]> {
+        const result = await this.projectManager.listProjects({})
+        return result.projects.map(project => ({
+            type: cloudService.AssetType.project,
+            title: project.name,
+            id: project.id,
+            parentId: '',
+            permissions: [],
+        }))
+    }
+
+    async listProjects(): Promise<cloudService.ListedProject[]> {
+        const result = await this.projectManager.listProjects({})
+        return result.projects.map(project => ({
+            name: project.name,
+            organizationId: '',
+            projectId: project.id,
+            packageName: project.name,
+            state: {
+                type: cloudService.ProjectState.created,
+            },
+            jsonAddress: null,
+            binaryAddress: null,
+        }))
+    }
+
+    async createProject(
+        body: cloudService.CreateProjectRequestBody
+    ): Promise<cloudService.CreatedProject> {
+        const project = await this.projectManager.createProject({
+            name: newtype.asNewtype<projectManager.ProjectName>(body.projectName),
+            ...(body.projectTemplateName ? { projectTemplate: body.projectTemplateName } : {}),
+            missingComponentAction: projectManager.MissingComponentAction.install,
+        })
+        return {
+            name: body.projectName,
+            organizationId: '',
+            projectId: project.projectId,
+            packageName: body.projectName,
+            state: {
+                type: cloudService.ProjectState.created,
+            },
+        }
+    }
+
+    async closeProject(projectId: cloudService.ProjectId): Promise<void> {
+        await this.projectManager.closeProject({ projectId })
+        this.currentlyOpenProject = null
+    }
+
+    async getProjectDetails(projectId: cloudService.ProjectId): Promise<cloudService.Project> {
```

Comment:
If getPtojectDetails returns `cloudService.Project`, it means that it can't get local projects info, right? This is surprising based on the naming `Backend.getProjectDetails` - there is no info in these names that it works with cloud projects only. (The same comment applies to other similar functions).

Reply:
`cloudService.Project` is just `Project`, it's defined in `cloudService` because it mirrors the return type from the cloud API. it's possible to rename the import (to something like `types` or `apiTypes`) to make this less confusing, if that's a satisfactory solution?